compiler/astalgo.nim

```diff
@@ -777,9 +777,10 @@ proc strTableInclReportConflict*(t: var TStrTable, n: PSym;
       replaceSlot = h
     h = nextTry(h, high(t.data))
   if replaceSlot >= 0:
+    result = t.data[replaceSlot] # found it
     if not onConflictKeepOld:
       t.data[replaceSlot] = n # overwrite it with newer definition!
-    return t.data[replaceSlot] # found it
+    return result
```

Comment:
this is not a refactoring, it's a change of semantics that wasn't mentioned in the PR; however, it's actually not a regression but a bugfix, see https://github.com/nim-lang/Nim/pull/18266

Reply:
It can happen that refactorings make things more correct; thanks for the testcase!